Chessington.GameEngine/Square.cs

```diff
@@ -49,5 +51,22 @@ public override string ToString()
         {
             return string.Format("Row {0}, Col {1}", Row, Col);
         }
+
+        public bool IsInbound()
+        {
+            return Enumerable.Range(0, 8).Contains(this.Col) && Enumerable.Range(0, 8).Contains(this.Row);
+        }
+
+        public bool IsOccupied(Board board)
+        {
+            if (board.GetPiece(this) == null)
+            {
+                return false;
```

Comment:
you don't need to return false or true. you can just return the statement in if clause